### DISCUSSION TITLE
如何表示剧集是否已经缓存过

### DISCUSSION THREAD

**Discussion by Him188** [open-ani/animeko#146] [Ideas]
急急急

该如何表示某一集是否已经有缓存呢

我觉得第一种有点怪, 第二三种很难看

![image](https://github.com/open-ani/ani/assets/12100985/b2735e05-a695-44c8-a059-8f3b9a687cbd)

![74982f7aafb93d3c30e4863e3c126a2f](https://github.com/open-ani/ani/assets/12100985/532929a2-bd46-4153-a638-53a349a1a67b)

![5a452c655b31464d87f972cc8d75d863](https://github.com/open-ani/ani/assets/12100985/f0c9e92c-4c9e-4701-b9f4-b4e9170a0292)

part of #145

**Comment by Him188** [comment]
目前我先临时写成第一种了

**Reply by NeKoOuO** [reply]
我反而覺得第二種沒那麼醜

顏色適配一下還會更好看

**Reply by NeKoOuO** [reply]
第一種實在太破壞結構了 強迫症看了很不舒服🫠

**Comment by Him188** [comment]
也不一定非要在这里展示缓存状态, 不过我觉得展示一下是比较好的 (至少要展示我即将要看的一集是否已经缓存了)

**Comment by NeKoOuO** [comment]
**可以在格子下面加一直線
用顏色區分狀態**

例如：
綠色已下載 紅色灰色未下載 中間色正在下載...等
或者用顏色深淺判別

**Reply by Him188** [reply]
确实, 好办法

**Reply by Him188** [reply]
尝试实现了一下感觉不简单, 颜色需要特别考虑
![image](https://github.com/open-ani/ani/assets/12100985/52df76f5-2da6-4d92-b307-92d73e88a693)

因为本身已经有了三种格子状态, 要能匹配上

**Reply by NeKoOuO** [reply]
是放在格子的下面捏🤔

**Reply by NeKoOuO** [reply]
應該不會跟原來的格子的顏色混到吧

**Reply by Him188** [reply]
![image](https://github.com/open-ani/ani/assets/12100985/2c1326bb-31af-4c03-b486-c6e6eb781ef7)
类似这样的效果?

颜色待会调一下

**Reply by Him188** [reply]
![image](https://github.com/open-ani/ani/assets/12100985/d85e44db-3109-41f3-8f91-67958dde0bdb)

**Reply by Him188** [reply]
![image](https://github.com/open-ani/ani/assets/12100985/4bf260b9-1cf9-41aa-aa1c-7cd07ea4aac9)

比较难支持可变主题+原本的格子颜色

**Reply by Him188** [reply]
就先按这个了，等以后再美化

**Reply by NeKoOuO** [reply]
可以耶 還不錯看

**Comment by NeKoOuO** [comment]
**把已下載在右上角放個星星**

![IMG_20240413_051255.jpg](https://github.com/open-ani/ani/assets/111511925/3beda98a-55d9-40ad-9f77-3b5a3cbf14dd)


未下載就普通樣子

單純區分有沒有下載的模式

**Reply by Him188** [reply]
这个要调颜色, 还会有边距问题

**Comment by NeKoOuO** [comment]
**更換圖標的形狀**

例如

從正方形變成圓角
從圓形變成圓角
從星形變成圓形...等等

**Reply by Him188** [reply]
感觉不太直观, 还得记忆形状对应的东西

**Comment by NeKoOuO** [comment]
有一個小問題

因為目前在下載那個頁面只有兩種圖標

分別是未下載的圖標和正在下載的圖標（轉圈圈的那個）

這樣有個問題就是如果是在排隊的一樣顯示未下載的圖標 這樣會造成不確定是否已經加入下載 導致重複點擊

我覺得要馬一樣改成圈圈 不然就是要再弄一個排隊中的圖標

**Reply by Him188** [reply]
排队状态应该只会持续几秒（这个状态实际上是在解析磁力链之类的）

**Reply by NeKoOuO** [reply]
> 排队状态应该只会持续几秒（这个状态实际上是在解析磁力链之类的）

我昨天自己一個個下載過

測試下來有些可能過了一分鐘還是跟未下載的樣子一樣（但列表裡面看是已經有了的）
